features/netsocket/PPPInterface.cpp

```diff
@@ -0,0 +1,193 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "PPPInterface.h"
+
+using namespace mbed;
+
+/* Interface implementation */
+PPPInterface::PPPInterface(PPP &ppp, OnboardNetworkStack &stack) :
+    _ppp(ppp),
+    _stack(stack),
+    _interface(NULL),
+    _dhcp(false),
+    _blocking(true),
+    _ip_address(),
+    _netmask(),
+    _gateway(),
+    _stream(NULL),
+    _ip_stack(DEFAULT_STACK),
+    _uname(),
+    _password()
+{
+}
+
+PPPInterface::~PPPInterface()
+{
+    _stack.remove_ppp_interface(&_interface);
+}
+
+nsapi_error_t PPPInterface::set_network(const char *ip_address, const char *netmask, const char *gateway)
+{
+    _dhcp = false;
+
+    strncpy(_ip_address, ip_address ? ip_address : "", sizeof(_ip_address));
+    _ip_address[sizeof(_ip_address) - 1] = '\0';
+    strncpy(_netmask, netmask ? netmask : "", sizeof(_netmask));
+    _netmask[sizeof(_netmask) - 1] = '\0';
+    strncpy(_gateway, gateway ? gateway : "", sizeof(_gateway));
+    _gateway[sizeof(_gateway) - 1] = '\0';
+
+    return NSAPI_ERROR_OK;
+}
+
+nsapi_error_t PPPInterface::set_dhcp(bool dhcp)
+{
+    _dhcp = dhcp;
+    return NSAPI_ERROR_OK;
+}
+
+nsapi_error_t PPPInterface::connect()
+{
+    _ppp.set_stream(_stream);
+    _ppp.set_ip_stack(_ip_stack);
+    _ppp.set_uname_password(_uname, _password);
+
+    if (!_interface) {
+        nsapi_error_t err = _stack.add_ppp_interface(_ppp, true, &_interface);
+        if (err != NSAPI_ERROR_OK) {
+            _interface = NULL;
+            return err;
+        }
+        _interface->attach(_connection_status_cb);
+    }
+
+    return _interface->bringup(_dhcp,
+                               _ip_address[0] ? _ip_address : 0,
+                               _netmask[0] ? _netmask : 0,
+                               _gateway[0] ? _gateway : 0,
+                               _ip_stack,
+                               _blocking);
+}
+
+nsapi_error_t PPPInterface::disconnect()
+{
+    if (_interface) {
+        return _interface->bringdown();
+    }
+    return NSAPI_ERROR_OK;
+}
+
+const char *PPPInterface::get_ip_address()
+{
+    if (_interface && _interface->get_ip_address(_ip_address, sizeof(_ip_address))) {
+        return _ip_address;
+    }
+
+    return NULL;
+}
+
+const char *PPPInterface::get_netmask()
+{
+    if (_interface && _interface->get_netmask(_netmask, sizeof(_netmask))) {
+        return _netmask;
+    }
+
+    return 0;
+}
+
+const char *PPPInterface::get_gateway()
+{
+    if (_interface && _interface->get_gateway(_gateway, sizeof(_gateway))) {
+        return _gateway;
+    }
+
+    return 0;
+}
+
+char *PPPInterface::get_interface_name(char *interface_name)
+{
+    if (_interface) {
+        return _interface->get_interface_name(interface_name);
+    }
+
+    return NULL;
+}
+
+void PPPInterface::set_as_default()
+{
+#if 0
```

Comment:
Please add comment why PPPINterface can't be set as default interface

Reply:
Removed the "if 0" and casting to lwip stack.